runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -149,38 +147,44 @@ func (d driver) Open(instanceID string, config map[string]any, client *activity.
 		return nil, err
 	}
 
-	var dsn string
+	// build clickhouse options
+	var opts *clickhouse.Options
 	if conf.DSN != "" {
-		dsn = conf.DSN
+		opts, err = clickhouse.ParseDSN(conf.DSN)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse DSN: %w", err)
+		}
 	} else if conf.Host != "" {
-		var dsnURL url.URL
-		dsnURL.Host = conf.Host
+		opts = &clickhouse.Options{}
+
+		// address
+		host := conf.Host
 		if conf.Port != 0 {
-			dsnURL.Host = fmt.Sprintf("%v:%v", dsnURL.Host, conf.Port)
+			host = fmt.Sprintf("%v:%v", conf.Host, conf.Port)
 		}
+		opts.Addr = append(opts.Addr, host)
```

Comment:
Why `append` instead of `=`?

Reply:
Yes that can be used as well. I copied this from the underlying driver to build options so used the same append code.